### ISSUE TITLE
New feature to download a file by assigning a string with a cookie if this is required.

### ISSUE THREAD

**Issue by sgou** [SimonSimCity/Xamarin-CrossDownloadManager#6]

**Comment by SimonSimCity** [comment]
Can you please describe what this feature should look like? Specially the phrase "if it is required" is quite vague ... like ... how can I determine if it's required?

You can add cookies like every other header to each of the DownloadFile instances by adding them to the instance before starting the download (`downloadManager.Start(file);`) or by passing it to the `downloadManager.CreateDownloadFile(url)` method.

An example for passing it to the method would look like this:

    var downloadManager = CrossDownloadManager.Instance;
    var file = downloadManager.CreateDownloadFile(url, new Dictionary<string, string> {
        { "Cookie", myCookie }
    });
    downloadManager.Start(file);

For cookies, you have to keep in mind, that the system-internal download manager may store these cookies and automatically sends them along in the next download-request as well. I have not tested that and am not aware of that it was mentioned in the documentations for either one (iOS or Android).

Making sure, the cookies are stored and sent with the next request, is, if you ask me  :wink:, not part of this plugin.

Closing the issue, as it's possible to send cookies. Please reopen this ticket if it isn't.
